### ISSUE TITLE
Invalid podSecurityPolicyTemplate  poisons Rancher's log 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#23931]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- run Rancher v2.3.1
- add a cluster with `Pod Security Policy Support` enabled 
- go to the `project/namespace` page 
- add a new project
- view the project in API
- catch the request for the action `setpodsecuritypolicytemplate`

<img width="734" alt="Screen Shot 2019-11-05 at 6 30 38 PM" src="https://user-images.githubusercontent.com/6218999/68260479-64d37f80-fffa-11e9-8b8e-6d4dd89d2e06.png">

- copy the command, change the value of `podSecurityPolicyTemplateId` to an unexisting one, for example 
```
'{"podSecurityPolicyTemplateId":"doesNotExist"}'
```

- run the command from the terminal

Results:
- no error is returned in the terminal
- in Rancher's log, the following error keeps showing up

`
2019/11/06 01:25:23 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-4x7mb/psptpb-6szfg [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "doesNotExist" not found
`

**Comment by jiaqiluo** [comment]
The bug fix is validated in v2.3.3-rc3

Steps: same as the issue
Result:
the following error returns
```
{"baseType":"error","code":"InvalidBodyContent","message":"podSecurityPolicyTemplate [restricted3] not found","status":422,"type":"error"}
```

**Comment by sowmyav27** [comment]
**verified on `2.3.3-rc3`**

**Steps:**
- Run steps as mentioned here - https://github.com/rancher/rancher/issues/23931#issue-518184091
- Error in the terminal -`{"baseType":"error","code":"InvalidBodyContent","message":"podSecurityPolicyTemplate [restr] not found","status":422,"type":"error"}`
- No error is seen in the rancher logs.
